File: ts/src/feature/base/BaseFeature.ts
import type { Feature, Context, FeatureOptions } from '../../types'




class BaseFeature implements Feature {
  version = '0.0.1'
  name = 'base'
  active = true


  init(_ctx: Context, _options: FeatureOptions): void | Promise<any> { }


  PostConstruct(this: any, _ctx: any) { }

  PostConstructEntity(this: any, _ctx: any) { }

  SetData(this: any, _ctx: any) { }

  GetData(this: any, _ctx: any) { }

  GetMatch(this: any, _ctx: any) { }


  PreTarget(this: any, _ctx: any) { }

  PreSpec(this: any, _ctx: any) { }

  PreRequest(this: any, _ctx: any) { }

  PreResponse(this: any, _ctx: any) { }

  PreResult(this: any, _ctx: any) { }

  PostOperation(this: any, _ctx: any) { }

}


export {
  BaseFeature
}

File: go/feature/base_feature.go
package feature

import (
	"voxgiguniversalsdk/core"
)

type BaseFeature struct {
	Version string
	Name    string
	Active  bool
}

func NewBaseFeature() *BaseFeature {
	return &BaseFeature{
		Version: "0.0.1",
		Name:    "base",
		Active:  true,
	}
}

func (f *BaseFeature) GetVersion() string { return f.Version }
func (f *BaseFeature) GetName() string    { return f.Name }
func (f *BaseFeature) GetActive() bool    { return f.Active }

func (f *BaseFeature) Init(ctx *core.Context, options map[string]any)  {}
func (f *BaseFeature) PostConstruct(ctx *core.Context)                 {}
func (f *BaseFeature) PostConstructEntity(ctx *core.Context)           {}
func (f *BaseFeature) SetData(ctx *core.Context)                       {}
func (f *BaseFeature) GetData(ctx *core.Context)                       {}
func (f *BaseFeature) GetMatch(ctx *core.Context)                      {}
func (f *BaseFeature) SetMatch(ctx *core.Context)                      {}
func (f *BaseFeature) PreTarget(ctx *core.Context)                     {}
func (f *BaseFeature) PreSelection(ctx *core.Context)                  {}
func (f *BaseFeature) PreSpec(ctx *core.Context)                       {}
func (f *BaseFeature) PreRequest(ctx *core.Context)                    {}
func (f *BaseFeature) PreResponse(ctx *core.Context)                   {}
func (f *BaseFeature) PreResult(ctx *core.Context)                     {}
func (f *BaseFeature) PreDone(ctx *core.Context)                       {}
func (f *BaseFeature) PreUnexpected(ctx *core.Context)                 {}
func (f *BaseFeature) PostOperation(ctx *core.Context)                 {}
